lib/puppet/provider/package/windows/exe_package.rb

```diff
@@ -55,7 +57,26 @@ def match?(resource)
     end
 
     def self.install_command(resource)
-      munge(resource[:source])
+      if Puppet::FileSystem.exist?(resource[:source])
+        munge(resource[:source])
+      else
+        begin
+          Puppet.debug ("The source %{resource_source} does not exist locally. Using it as URI ") % { resource_source: resource[:source]}
+          uri = URI.parse(resource[:source])
+        rescue URI::InvalidURIError => e
+          Puppet.debug(_("URI is invalid: %{detail}") % { detail: e.message}, e)
+        rescue => detail
+          raise Puppet::Error.new(_("Error when parsing URI %{detail}") % { detail: detail.message}, detail)
+        end
+        require 'open-uri'
+
+        file_location = File.join(Dir.tmpdir() ,File.basename(resource[:source]))
+        File.open(file_location, "wb") do |file|
+          file.write(URI.open(uri).read)
+        end
```

Comment:
`open-uri` should be avoided because we don't have control over the certs that it uses. Better to use puppet's httpclient and stream the response body (since the package could be large):

```ruby
client = Puppet.runtime[:http]
response = client.get(uri, options: {include_system_store: true})
raise ... unless response.success?

File.open(file_location, "wb") do |file|
  response.read_body do |data|
    file.write(data)
  end
end
```

The 'include_system_store' allows puppet to trust both the puppet CA(s) and CA certs set via `SSL_CERT_DIR` and `SSL_CERT_FILE` environment variables. When running from puppet-agent packages, those variables point to our vendored CA bundle (https://github.com/puppetlabs/puppet-ca-bundle)